### PR TITLE
feat(Rich Text Editor): Remark plate inline

### PR DESCRIPTION
**Pull request by bojangles-m**
Copied code from: https://github.com/hanford/remark-slate into project to use it to transform Markdown into Slate and Slate to DM.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -161,6 +161,7 @@
     },
     "peerDependencies": {
         "react": "^17.0.2",
-        "react-dom": "^17.0.2"
+        "react-dom": "^17.0.2",
+        "escape-html": "^1.0.3"
```
Comment (SamuelAlev) [inline package.json:165]: It is a dependency

File: src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    };
+    emphasis_mark: string;
+    strong_mark: string;
+    delete_mark: string;
+    inline_code_mark: string;
+    thematic_break: string;
+    image: string;
+}
+
+export type MdAstNodeType =
```
Comment (SamuelAlev) [inline src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts:27]:
```suggestion
export type MarkdownAstNodeType =
```

File: package.json (hunk 1 above)
Comment (bojangles-m) [reply package.json:165]: Updated

File: src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts (hunk 2 above)
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts:27]: Updated

File: src/components/RichTextEditor/serializer/md/RemarkPlate/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { MdAstNode, OptionType } from './astTypes';
```
[the remark this answers is not in the document]
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/md/RemarkPlate/plugin.ts:1]: Updated

File: src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts [hunk 4]
```diff
@@ -0,0 +1,210 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
```
[the remark this answers is not in the document]
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts:1]: Moved

File: src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+export interface InputNodeTypes {
```
[the remark this answers is not in the document]
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts:3]: Updated

[the remark this answers is not in the document]
Comment (bojangles-m) [reply src/components/RichTextEditor/serializer/md/RemarkPlate/astTypes.ts]: All interfaces updated with types

**Review by SamuelAlev — APPROVED** [review]

**Pull request merged**